src/app/issueForm/issueForm.controller.ts

```diff
@@ -18,45 +18,83 @@ module testApp {
   export class IssueFormController {
     public awesomeThings: Thing[];
     public search: string;
-    private pendingTask: any;
-    private $http: any;
-    private $window: any;
-    private $stateParams: any;
+    public mode;
+    public issueNumber: number;
     public repository;
     public owner;
     public repo;
+    public issue: any;
+
+    private $http: any;
+    private HttpService: any;
+    private $window: any;
+    private $stateParams: any;
+    private $state: any;
 
     submitForm(isValid, issue) {
       if(!isValid) {
         alert("Not valid");
       } else {
         var authToken;
-        var _this = this;
         var params = {};
         params['title'] = issue.title;
         params['body'] = issue.body;
-        this.$http.post(
+        var promise = null;
+
+        if(this.mode){
+          promise = this.HttpService.call('PATCH', "repos/" + this.owner + "/" + this.repo +  "/issues/" + this.issueNumber, params)
```

Comment:
I think it would be better to have something like this:

```typescript
promise = this.repository.updateIssue(this.owner, this.repo, this.issueNumber, issue);
```

Internally, `repository` (which is a service), would call the HttpService like this:
```typescript

public updateIssue(owner: string, repoName: string, issue: any) : ng.IPromise<any> {
  let params = { title: issue.title, body: issue.body };
  return this.HttpService.call('PATCH', "repos/" + owner + "/" + repoName +  "/issues/" + issue.id, params);
}
```